### PR TITLE
Throw Asserts when enums overflow.

### PR DESCRIPTION
**Pull request by GuyMcComb**

### CODE REVIEW

File: src/cairoUtilFuncGen/enumInputCheck.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      name,
+      [['arg', typeNameFromTypeNode(inputType, this.ast), DataLocation.Default]],
+      [['ret', typeNameFromTypeNode(nodeType, this.ast), DataLocation.Storage]],
+      ['range_check_ptr'],
+      this.ast,
+      nodeInSourceUnit ?? nodeInput,
+      FunctionStateMutability.Pure,
+      FunctionStubKind.FunctionDefStub,
+    );
+    const functionInput = cloneASTNode(nodeInput, this.ast);
```
Comment (DomHenderson) [inline src/cairoUtilFuncGen/enumInputCheck.ts:41]: Shouldn't be any need to clone here I don't think

File: src/cairoUtilFuncGen/enumInputCheck.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    this.ast.setContextRecursive(functionInput);
+    return createCallToFunction(functionStub, [functionInput], this.ast);
+  }
+
+  private getOrCreate(type: TypeNode, enumDef: EnumDefinition): string {
+    const key = type.pp();
```
Comment (DomHenderson) [inline src/cairoUtilFuncGen/enumInputCheck.ts:48]: key needs to be more specific than this. Needs to include both the enum and the input type

File: src/cairoUtilFuncGen/enumInputCheck.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  ): FunctionCall {
+    const nodeType = getNodeType(node, this.ast.compilerVersion);
+    const inputType = getNodeType(nodeInput, this.ast.compilerVersion);
+
+    this.sourceUnit = this.ast.getContainingRoot(nodeInSourceUnit);
+    const name = this.getOrCreate(inputType, enumDef);
+    const functionStub = createCairoFunctionStub(
+      name,
+      [['arg', typeNameFromTypeNode(inputType, this.ast), DataLocation.Default]],
+      [['ret', typeNameFromTypeNode(nodeType, this.ast), DataLocation.Storage]],
```
Comment (DomHenderson) [inline src/cairoUtilFuncGen/enumInputCheck.ts:34]: Why storage?

**Review by DomHenderson — CHANGES_REQUESTED** [review]
This is more what I meant yes

File: src/cairoUtilFuncGen/enumInputCheck.ts (hunk 1 above)
Comment (GuyMcComb) [reply src/cairoUtilFuncGen/enumInputCheck.ts:41]: Ahh yes good point.

File: src/cairoUtilFuncGen/enumInputCheck.ts (hunk 3 above)
Comment (GuyMcComb) [reply src/cairoUtilFuncGen/enumInputCheck.ts:34]: Because Enums are stored in storage aren't they?

Comment (DomHenderson) [reply src/cairoUtilFuncGen/enumInputCheck.ts:34]: no more than ints are

**Review by DomHenderson — APPROVED** [review]

**Pull request merged**